Rakefile

```diff
@@ -452,36 +424,56 @@ task :test do
         "service redis-flapjack start",
         "service flapjack start"
       ]
-      image = "#{pkg.distro}:#{pkg.distro}#{pkg.distro_release}"
     end
+    OmnibusFlapjack::Helpers.run_tests_in_docker(options)
+  end
+end
 
-    test_cmd << [
-      "cd /mnt/omnibus-flapjack",
-      "gem install bundler --no-ri --no-rdoc",
-      "bundle",
-      "bundle exec rspec spec/serverspec"
-    ]
-
-    test_cmd = test_cmd.flatten.join(" && ")
-
-    container_name = "flapjack-test-#{pkg.distro_release}"
-
-    docker_cmd_string = [
-      'docker', 'run', '-t',
-      '--attach', 'stdout',
-      '--attach', 'stderr',
-      '--name', container_name,
-      "-v #{Dir.pwd}:/mnt/omnibus-flapjack",
-      "#{image}", 'bash', '-l', '-c',
-      "\'#{test_cmd}\'"
-    ].join(" ")
-    puts "Executing: " + docker_cmd_string
-    unless dry_run
-      OmnibusFlapjack::Helpers.run_docker(docker_cmd_string)
-      puts "Purging the container #{container_name}"
-      Mixlib::ShellOut.new("docker rm #{container_name}").run_command
+desc "Test a flapjack package from the repository, using docker"
+task :post_publish_test do
+  # Choose distro & release
+  # Clone vagrant flapjack
+  # Mount vagrant-flapjack in docker container
+  # Start up Docker
+  # Install puppet
+  # Run puppet from vagrant-flapjack
+  # Run serverspec
+  # Run capybara on supported OSes
+
+  distro         = ENV['DISTRO']
+  distro_release = ENV['DISTRO_RELEASE']
+  component      = ENV['FLAPJACK_COMPONENT']
+  if distro == 'centos'
+    component = component == 'main' ? 'flapjack' : 'flapjack-experimental'
+  end
+  arch = distro == 'centos' ? 'x86_64' : 'amd64'
+
+  unless dry_run
+    if File.exist?('vagrant-flapjack/Vagrantfile')
+      Mixlib::ShellOut.new("cd vagrant-flapjack && git checkout test-deps-in-puppet && git pull && cd -", :live_stream => $stdout).run_command.error!
```

Comment:
test-deps-in-puppet is a branch that installs poltergeist via a puppet module for now.  This will grow a number of other dependencies that are only needed for capybara in future.